lib/services/firestore_service.dart

```diff
@@ -37,4 +38,28 @@ class FirestoreService {
     }
     return taskList;
   }
+
+  Future<List<Task>> getListofNewTasks() async {
+    List<Task> newTaskList=[];
```

Comment:
This is going to slow down the system a lot, since we are synchronously adding it.
https://api.dart.dev/stable/2.5.0/dart-async/Future/wait.html

You should be able to create List<Future<Task>> and then wait for all futures 
Future.wait(list<future<task>>) and return that instead.